parquet-pig/src/main/java/org/apache/parquet/pig/ParquetLoader.java

```diff
@@ -380,4 +397,133 @@ private Schema getSchemaFromRequiredFieldList(Schema schema, List<RequiredField>
+        Expression.OpType.OP_GE,
+        Expression.OpType.OP_LT,
+        Expression.OpType.OP_LE,
+        Expression.OpType.OP_AND,
+        Expression.OpType.OP_OR,
+    };
+
+    return Arrays.asList(supportedTypes);
+  }
+
+  @Override
+  public void setPushdownPredicate(Expression e) throws IOException {
+    LOG.info("Pig pushdown expression: " + e);
+
+    FilterPredicate pred = buildFilter(e);
+    LOG.info("Parquet filter predicate expression: " + pred);
+
+    storeInUDFContext(ParquetInputFormat.FILTER_PREDICATE, pred);
+  }
+
+  private FilterPredicate buildFilter(Expression e) {
+    if (e instanceof BinaryExpression) {
+      Expression lhs = ((BinaryExpression) e).getLhs();
+      Expression rhs = ((BinaryExpression) e).getRhs();
+
+      switch (e.getOpType()) {
+        case OP_EQ: return eq(getColumn((Column) lhs), getValue(rhs));
+        case OP_GT: return gt(getColumn((Column) lhs), getValue(rhs));
+        case OP_GE: return gtEq(getColumn((Column) lhs), getValue(rhs));
+        case OP_LT: return lt(getColumn((Column) lhs), getValue(rhs));
+        case OP_LE: return ltEq(getColumn((Column) lhs), getValue(rhs));
+        case OP_AND: return and(buildFilter(lhs), buildFilter(rhs));
+        case OP_OR: return or(buildFilter(lhs), buildFilter(rhs));
+      }
+    }
+
+    return null;
+  }
+
+  private ColumnWrapper getColumn(Column c) {
+    String column = c.getName();
+
+    try {
+      FieldSchema f = schema.getField(column);
+
+      switch (f.type) {
+        case DataType.BOOLEAN: return new ColumnWrapper<Binary>(binaryColumn(f.alias));
+        case DataType.INTEGER: return new ColumnWrapper<Integer>(intColumn(f.alias));
+        case DataType.LONG: return new ColumnWrapper<Long>(longColumn(f.alias));
+        case DataType.FLOAT: return new ColumnWrapper<Float>(floatColumn(f.alias));
+        case DataType.DOUBLE: return new ColumnWrapper<Double>(doubleColumn(f.alias));
+        case DataType.CHARARRAY: return new ColumnWrapper<Binary>(binaryColumn(f.alias));
+      }
+
+    } catch (FrontendException e) {
+      throw new RuntimeException("Error processing pushdown for column:" + c , e);
+    }
+
+    return null;
+  }
+
+  private static class ColumnWrapper<T extends Comparable<T>> extends Operators.Column<T> implements Operators.SupportsLtGt, Serializable {
+
+    Operators.Column<T> wrapped;
+
+    public ColumnWrapper(Operators.Column<T> wrapped) {
+      super(wrapped.getColumnPath(), wrapped.getColumnType());
+      this.wrapped = wrapped;
+    }
+
+    @Override
+    public Class<T> getColumnType() {
+      return wrapped.getColumnType();
+    }
+
+    @Override
+    public ColumnPath getColumnPath() {
+      return wrapped.getColumnPath();
+    }
+  }
+
+  private Comparable getValue(Expression expr) {
+    switch(expr.getOpType()) {
+      case TERM_COL:
+        //return ((Column) expr).getName();
+      case TERM_CONST:
+        Comparable value = (Comparable) ((Const) expr).getValue();
```

Comment:
If that's the case, then I don't think this method should be called at all. Parquet doesn't support pushdown comparison between columns, so I think the right thing to do is to check if it is a `Const` and return null from `buildFilter` instead of calling this and causing a `RuntimeException`.